src/main/java/seedu/club/ui/PollListPanel.java

```diff
@@ -23,53 +22,62 @@
 public class PollListPanel extends UiPart<Region> {
     private static final String FXML = "PollListPanel.fxml";
     private final Logger logger = LogsCenter.getLogger(PollListPanel.class);
+    private boolean isDisplayingPollResults;
+    private ObservableList<Poll> pollList;
 
     @FXML
     private ListView<PollCard> pollListView;
 
     public PollListPanel(ObservableList<Poll> pollList) {
         super(FXML);
-        setConnections(pollList);
+        this.pollList = pollList;
+        setPollListView();
         registerAsAnEventHandler(this);
     }
 
-
-    private void setConnections(ObservableList<Poll> pollList) {
-        setPollListView(pollList);
-        setEventHandlerForSelectionChangeEvent();
-    }
-
-    private void setPollListView(ObservableList<Poll> pollList) {
+    private void setPollListView() {
         ObservableList<PollCard> mappedList = EasyBind.map(
-                pollList, (poll) -> new PollCard(poll, pollList.indexOf(poll) + 1));
+                pollList, (poll) -> {
+                if (isDisplayingPollResults) {
+                    return new PollCard(poll, pollList.indexOf(poll) + 1);
+                } else {
+                    return new RestrictedPollCard(poll, pollList.indexOf(poll) + 1);
+                }
+            });
         pollListView.setItems(mappedList);
         pollListView.setCellFactory(listView -> new PollListViewCell());
     }
 
-    private void setEventHandlerForSelectionChangeEvent() {
-        pollListView.getSelectionModel().selectedItemProperty()
-                .addListener((observable, oldValue, newValue) -> {
-                    if (newValue != null) {
-                        logger.fine("Selection in poll list panel changed to : '" + newValue + "'");
-                        raise(new PollPanelSelectionChangedEvent(newValue));
-                    }
-                });
+    /**
+     * Shows results of polls
+     */
+    protected void showPollResults() {
```

Comment:
any reason why the access is `protected` ?

Reply:
Ops. Initially thought was gonna call from out of class. Forget change to private. Thanks 😄